File: server/order-service/src/main/java/com/sudo248/orderservice/controller/order/dto/UpsertUserProductDto.java
package com.sudo248.orderservice.controller.order.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@Data
@AllArgsConstructor
public class UpsertUserProductDto {
    private String productId;
    private String userId;

    public UpsertUserProductDto(String productId) {
        this.productId = productId;
        this.userId = null;
    }
}

File: server/product-service/src/main/kotlin/com/sudoo/productservice/dto/UpsertUserProductDto.kt
package com.sudoo.productservice.dto

import java.time.LocalDateTime

data class UpsertUserProductDto(
    val userProductId: String? = null,
    val productId: String? = null,
    val userId: String? = null,
    val rate: Float? = null,
    val comment: String? = null,
    var createdAt: LocalDateTime? = null,
    var images: List<String>? = null,
)